lib/ruby/truffle/truffle/bigdecimal.rb

```diff
@@ -181,6 +189,70 @@ def frac
       self
     end
   end
+
+  def to_s(format = 'E')
+    if finite?
+      spaces       = format.to_i
+      prefix       = case
+                     when self > 0
+                       if [' ', '+'].include? format[0]
+                         format[0]
+                       else
+                         ''
+                       end
+                     when self < 0
+                       '-'
+                     when self == 0
+                       ''
+                     end
+      float_format = format[-1] == 'F'
+
+      unscaled_value = unscaled
+      exponent_value = self.exponent
+
+      if float_format
+        before_dot, after_dot = case
+                                when exponent_value > unscaled_value.size
+                                  [unscaled_value + '0' * (exponent_value - unscaled_value.size),
+                                   '0']
+                                when exponent_value <= 0
+                                  ['0',
+                                   '0' * exponent_value.abs + unscaled_value]
+                                else
+                                  after_dot = unscaled_value[exponent_value..-1]
+
+                                  [unscaled_value[0...exponent_value],
+                                   after_dot.empty? ? '0' : after_dot]
+                                end
```

Comment:
I tend to prefer case over of/elsif in when there is more than two branches to distinguish it clearly from simple if. 

Multi-assign is used here to be DRY and "functional". In the current arrangement it is clear that whole block is about assigning those variables, but if the variables are in the case (or if/elsif) a reader has to read whole block to determine that it's about assigning of those variables and nothing else.

Anyway this is just stylistic issue, I don't mind changing it if you think this is too weird.